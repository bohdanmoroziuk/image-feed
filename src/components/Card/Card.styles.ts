import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  image: {
    aspectRatio: 1,
    backgroundColor: 'rgba(0, 0, 0, 0.02)',
  }
});
