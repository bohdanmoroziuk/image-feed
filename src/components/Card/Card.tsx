import { FunctionComponent } from 'react';
import { View, Image, ImageSourcePropType } from 'react-native';

import AuthorRow from 'src/components/AuthorRow';

import styles from './Card.styles';

export interface CardProps {
  fullName: string;
  image: ImageSourcePropType;
  linkText: string;
  onPressLinkText: () => void;
}

export const Card: FunctionComponent<CardProps> = ({
  fullName,
  image,
  linkText,
  onPressLinkText,
}) => (
  <View>
    <AuthorRow
      fullName={fullName}
      linkText={linkText}
      onPressLinkText={onPressLinkText}
    />
    <Image
      style={styles.image}
      source={image}
    />
  </View>
);