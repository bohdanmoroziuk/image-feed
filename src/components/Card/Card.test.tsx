import { render, waitFor } from '@testing-library/react-native';

import { Card } from './Card';

describe('Card', () => {
  it('renders successfully', async () => {
    const { toJSON } = render(
      <Card 
        fullName="First Last"
        linkText="Comments"
        image={{ uri: 'https://unsplash.it/600/600' }}
        onPressLinkText={() => {
          console.log('Pressed link!');
        }}
      />
    );

    await waitFor(() => {
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
