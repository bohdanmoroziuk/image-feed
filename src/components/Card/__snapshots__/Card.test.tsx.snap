// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders successfully 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 50,
        "paddingHorizontal": 10,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#0b0b9e",
            "borderRadius": 17.5,
            "height": 35,
            "width": 35,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "white",
          }
        }
      >
        FL
      </Text>
    </View>
    <Text
      numberOfLines={1}
      style={
        Object {
          "flex": 1,
          "marginHorizontal": 6,
        }
      }
    >
      First Last
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        numberOfLines={1}
      >
        Comments
      </Text>
    </View>
  </View>
  <Image
    source={
      Object {
        "uri": "https://unsplash.it/600/600",
      }
    }
    style={
      Object {
        "aspectRatio": 1,
        "backgroundColor": "rgba(0, 0, 0, 0.02)",
      }
    }
  />
</View>
`;
